app/src/main/java/com/example/todolistproject/view/TaskActivity.kt

```diff
@@ -0,0 +1,243 @@
+                    openDeleteTaskDialog(getItem(position))
+                }
+
+                override fun onTaskCompleteClick(position: Int) {
+                    setTaskCompleted(getItem(position))
+                }
+            }
+        }
+
+        binding.rvTodoList.run {
+            layoutManager = LinearLayoutManager(this@TaskActivity)
+            adapter = mAdapter
+        }
+    }
+
+    private fun observeViewModel() {
+        mViewModel.getTaskList().observe(this, {
+            mAdapter.setTaskItems(it)
+            showTaskList(true)
+        })
+    }
+
+    fun openAddTaskDialog() {
+        val dialogViewBinding = DialogAddTaskBinding.inflate(LayoutInflater.from(this))
+
+        val dialog =
+            AlertDialog.Builder(this).setTitle(getString(R.string.todo_add_task))
+                .setView(dialogViewBinding.root)
+                .setPositiveButton(getString(R.string.todo_add)) { dialog, _ ->
+                    addTask(dialogViewBinding)
+                    dialog.dismiss()
+                }
+                .setNegativeButton(getString(R.string.todo_cancel)) { dialog, _ -> dialog.dismiss() }
+                .create()
+
+        dialogViewBinding.startDateEditText.inputType = InputType.TYPE_NULL
+        dialogViewBinding.startDateEditText.setOnClickListener {
+            openDatePickerDialog(dialogViewBinding.startDateEditText)
+        }
+        dialogViewBinding.endDateEditText.inputType = InputType.TYPE_NULL
+        dialogViewBinding.endDateEditText.setOnClickListener {
+            openDatePickerDialog(dialogViewBinding.endDateEditText)
+        }
+        dialog.show()
+    }
+
+    private fun addTask(dialogViewBinding: DialogAddTaskBinding) {
+        val title = dialogViewBinding.taskTitleEditText.text.toString()
+        val description = dialogViewBinding.taskDescriptionEditText.text.toString()
+        val startDate = dialogViewBinding.startDateEditText.text.toString()
+        val endDate = dialogViewBinding.endDateEditText.text.toString()
+
+        val task =
+            Task(
+                id = null,
+                isCompleted = false,
+                title = title,
+                description = description,
+                startDate = startDate,
+                endDate = endDate
+            )
+        mViewModel.insertTask(task)
+    }
+
+    private fun openModifyTaskDialog(task: Task) {
+        val dialogViewBinding = DialogAddTaskBinding.inflate(LayoutInflater.from(this))
+        dialogViewBinding.task = task
+
+        val dialog = AlertDialog.Builder(this).setTitle(getString(R.string.todo_edit_task))
+            .setView(dialogViewBinding.root)
+            .setPositiveButton(getString(R.string.todo_okay)) { dialog, _ ->
+                modifyTask(dialogViewBinding, task)
+                dialog.dismiss()
+            }
+            .setNegativeButton(getString(R.string.todo_cancel)) { dialog, _ ->
+                dialog.dismiss()
+            }
```

Comment:
```suggestion
            .setPositiveButton(getString(R.string.todo_okay)) { dialog, _ ->
                modifyTask(dialogViewBinding, task)
            }
            .setNegativeButton(getString(R.string.todo_cancel), null)
```
대부분의 SDK에서는 기본 컨트롤버튼인 ok cancel등에 dismiss기능이 기본으로 재공 되요.